server/src/__tests__/utils/mikro-integration-test-environment.ts

```diff
@@ -0,0 +1,56 @@
+import { mocks } from '../../test-utils'
+import { MikroDB, setupMikroDB } from '../../setup/db'
+import { setupApplication } from '../../setup/application'
+import { App, setupWaterparkExpress } from '../../setup/http/express'
+import { ClubEntity } from '../../shared/infra/db/entities/club.entity'
+import { EventEntity } from '../../shared/infra/db/entities/event.entity'
+import { TagEntity } from '../../shared/infra/db/entities/tags/tag.entity'
+
+export interface Variables {
+  app: App
+}
+
+// This is similar to Jest environments.
+// We don't use Jest environments here because they modify the global object in a type-unsafe way.
+class MikroIntegrationTestEnvironment {
+  protected mikroDB!: MikroDB
+
+  protected ids!: Array<string>
+  protected clubEntities!: Array<ClubEntity>
+  protected eventEntities!: Array<EventEntity>
+  protected tagEntities!: Array<TagEntity>
+
+  public async setup(): Promise<Variables> {
+    this.mikroDB = await setupMikroDB({ debug: false })
+    const { orm, entityRepos, repos } = this.mikroDB
+
+    this.ids = [1, 2, 3].map(String)
+    this.clubEntities = this.ids.map(mocks.mockClubEntity)
```

Comment:
ooh, subclasses are a great idea!

At the very least, I shouldn't assume that adding those entities is part of the _base_ integration test environment, it should be in some subclass. I'll factor that into a subclass in this PR, and that'll also serve as a signal to future devs that subclasses are encouraged 😄

Reply:
Every time Justin kills a class 5 more appear. 

Not sure if we should include in this PR. If it's definitely part of IoC then go ahead, but I think we can have further breakdown of the testing environment in a separate PR. Just my two cents.

Reply:
I think it is in the scope of this PR because this is actually the PR that introduced the testing environment in the first place. (The old `DB` object doesn't count as an environment.) I might as well get it right the first time.